chia/types/blockchain_format/program.py

```diff
@@ -136,11 +136,34 @@ def curry(self, *args) -> "Program":
             fixed_args = [4, (1, arg), fixed_args]
         return Program.to([2, (1, self), fixed_args])
 
-    def uncurry(self) -> Tuple["Program", "Program"]:
-        r = uncurry(self)
-        if r is None:
+    def uncurry(self) -> Tuple[Program, Program]:
+        try:
+            i = self.as_iter()
+            ev = next(i)
+            # the first instruction is eval
+            if ev.atom != b"\x02":
+                return self, self.to(0)
+            quoted_inner = next(i)
+            # then the quoted inner mod
+            if quoted_inner.pair[0].atom != b"\x01":
+                return self, self.to(0)
+            mod = quoted_inner.pair[1]
+            uncurried_args = next(i)
```

Comment:
notably, I'm not checking that `i` is now at the end. I get the impression the only way to do that is to call `next()` again and expect a `StopIteration` exception. Is it important to ensure there's no garbage at the end of this list?

The same thing goes for the `ai` iterator below.